app/services/data/update-claim-overpayment-status.js

```diff
@@ -2,18 +2,31 @@ const config = require('../../../knexfile').intweb
 const knex = require('knex')(config)
 
 const insertClaimEvent = require('./insert-claim-event')
+const overpaymentActionEnum = require('../../constants/overpayment-action-enum')
+
+const newLine = '\r\n'
 
 module.exports = function (claim, overpaymentResponse) {
-  var eventLabel = overpaymentResponse.isOverpaid ? 'OVERPAID-CLAIM' : 'OVERPAID-CLAIM-RESOLVED'
-  var note = overpaymentResponse.isOverpaid ? overpaymentResponse.reason : ''
+  var eventLabel = overpaymentResponse.action
+  var note = overpaymentResponse.reason
+  var toBeMarkedAsOverpaid = overpaymentResponse.action === overpaymentActionEnum.OVERPAID
+  var toBeResolved = overpaymentResponse.action === overpaymentActionEnum.RESOLVE
+  var toBeUpdated = overpaymentResponse.action === overpaymentActionEnum.UPDATE
+
   var updateClaim = {
-    IsOverpaid: overpaymentResponse.isOverpaid
+    IsOverpaid: toBeMarkedAsOverpaid || toBeUpdated
   }
 
-  if (overpaymentResponse.isOverpaid === true) {
+  if (toBeMarkedAsOverpaid) {
     updateClaim.OverpaymentAmount = overpaymentResponse.amount
     updateClaim.RemainingOverpaymentAmount = overpaymentResponse.amount
     updateClaim.OverpaymentReason = overpaymentResponse.reason
+  } else if (toBeResolved) {
+    updateClaim.RemainingOverpaymentAmount = overpaymentResponse.remaining
+  } else if (toBeUpdated) {
```

Comment:
Currently, I don't think so.  The action field gets set in the overpayment-response domain object in the function 'getOverpaymentStatus'.  This function will always return one of three values so unless that function is changed to return null, or someone creates their own overpayment-response object manually, you should never fall out of this conditional statement